Control/Newtype/Generics/Generic.hs

```diff
@@ -0,0 +1,211 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+#if __GLASGOW_HASKELL__ < 710
+-- I guess GHC decides better these days.
+{-# LANGUAGE UndecidableInstances       #-}
+#endif
+
+{- |
+A version of the 'Newtype' typeclass and related functions.
+Primarily pulled from Conor McBride's Epigram work. Some examples:
+
+>>> ala Sum foldMap [1,2,3,4]
+10
+
+>>> ala Endo foldMap [(+1), (+2), (subtract 1), (*2)] 3
+8
+
+>>> under2 Min (<>) 2 1
+1
+
+>>> over All not (All False)
+All {getAll = True)
+
+There are neat things you can do with this with /any/ 'Generic' newtype.
+
+ > {-# LANGUAGE DeriveGeneric #-}
+ > import GHC.Generics
+ >
+ > (...)
+ > newtype Example = Example Int
+ >   deriving (Generic)
+ >
+ > instance Newtype Example
```

Comment:
This section seems to suggest that writing this instance is required. Let's maybe instead show a short example that demonstrates how to use this module with a custom type.

Something like

```
>>> :set -XDeriveGeneric
>>> import GHC.Generics
>>> newtype Example = Example Int deriving Generic
>>> op Example (Example 3)
>>> 3
```